pkg/collector/horizontalpodautoscaler.go

```diff
@@ -86,11 +86,17 @@ func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1al
 			},
 		}
 
-		if otelcol.Spec.Autoscaler != nil && otelcol.Spec.Autoscaler.Behavior != nil {
+		if otelcol.Spec.Autoscaler.Behavior != nil {
```

Comment:
have we confirmed that Autoscaler isn't nil by this point?

Reply:
Hmm there are several references of using `otelcol.Spec.Autoscaler` without a nil check in this function. In fact this is the only place that nil checks. If it is unset, `otelcol.Spec.Autoscaler` is created by the defaulting webhook at https://github.com/open-telemetry/opentelemetry-operator/blob/main/apis/v1alpha1/opentelemetrycollector_webhook.go#L72

So it should never be nil, but now I'm wondering if there cases where the defaulting webhook is never called and this causes a segfault?

Reply:
thats a good q... maybe we add the guard at the start of the function?

Reply:
Added the nil check at the start and returning nil if it is unset!